### ISSUE TITLE
Alpha button

### ISSUE THREAD

**Issue by origami-z** [jpmorganchase/salt-ds#4280]
Follow up from [#4188](https://github.com/jpmorganchase/salt-ds/issues/4188), add button to work with System Status

**Comment by mark-tate** [comment]
Latte Goal: finish spec

**Comment by mark-tate** [comment]
@jake-costa we need some a11y help on this

**Comment by jake-costa** [comment]
@mark-tate working with @dplsek to go over what assistance is required for this. Working to clarify Non-text Contrast 1.4.11.

**Comment by dplsek** [comment]
I reviewed the design proposal with @bhoppers2008 @joshwooding @jake-costa and they achieve ADA and design requirements. 

Got a preliminary 'thumbs up' from @origami-z via email who could not make the call.

Specs for [Salt Legacy](https://www.figma.com/design/6LVMkWiMGpBBcI60qWltoV/Alpha-Button-(Legacy)?node-id=0-1&t=Bpd7kEYtIyWjPCHv-1)
Specs for [Salt Next](https://www.figma.com/design/qG5IV24qJfxrf3GA1bv972/Alpha-Button-(Next)?node-id=2024-1091&t=C3q4GsS5a51vKI16-1)

Still need to review this in detail with @pseys and @origami-z to confirm if this is technically feasible, can be wired systematically, and the necessary tokens to do so.

**Comment by mark-tate** [comment]
Lungo update: reviewed, proposal for implementation by @origami-z

**Comment by origami-z** [comment]
Thoughts from meeting earlier: 

We saw 2 sets of alpha buttons

1. mode agnostic for 500 background
2.flipped for standard background variants (white/black texts)

1st one is needed for unblocking system status. 2nd one is preferred visual for replacing transparent button within Banner.

As soon as we release the button for system status, the community will ask to use them on other places (e.g. solid background toast, even we don't provide them). So limiting it to be `SystemStatusButton` won't be a good choice. However, if we don't have time to work on all buttons right now, we can opt for staged approach

1. release `SystemStatusButton`, constraint for only `SystemStatus` usage
2. [future] release `AlphaButton` (or other name), with 2 variants mentioned above, and swap internal implementation of `SystemStatusButton` to the new button

For alpha button, we need to think about

- Good name for the 2 variants 
- Recommendation when used outside of Salt (e.g. BLOM)

Additional question for active state for @bhoppers2008 and @jake-costa, active state are usually transient state when used standalone. What about when the alpha button is used as menu trigger and when menu is open (active state will be kept there). We can't prevent users to pair the new button with overlay or menu for additional workflow.

**Comment by origami-z** [comment]
> Does it cause us any complications down the line if we create a `systemStatusButton` component and then rename it?

Rename only works at major breaking change, which would be tied to a specific future event. It's better we plan something not relying on those?

**Comment by pseys** [comment]
Yes I was thinking it makes more sense that we get the naming right from the start, rather than try to rename it down the road.

**Comment by origami-z** [comment]
> Yes I was thinking it makes more sense that we get the naming right from the start, rather than try to rename it down the road.

Yes, it could work. Do we know a good component name (and potential variant name) we can go to directly? With the complexity of mode agnostic / light+dark + potential different background? I'm not sure whether "AlphaButton" is a safe bet....

**Comment by origami-z** [comment]
My proposal below, feedback welcome

1. Create a `GhostButon`, with prop `adaptiveAppearance = "solid" | "non-solid"`, default to "non-solid", to be paired with most default option of "container" type of components
1. Deprecate `variant` in Banner, replace with `appearance = "solid" | "filled" | "bordered"`, default to "bordered" for backwards compatibility.  New `filled` is introduced to the overall appearance options, to describe design choice of border + light-weight filled background (prior primary/secondary).
1. Deprecate `SystemStatus`, replace with `<Banner appearance="filled">`, move relevant use case and guidance to pattern section

See https://github.com/jpmorganchase/salt-ds/issues/4280#issuecomment-2482426232 above, and more notes below

- Explored options to add "ghost" prop to existing Button component, but it's hard to explain usage guidance when paired with sentiment / appearance options 
- Explored options to add "ghost" to Button's appearance prop option, similar to above, can't really explain the relationship with sentiment, and can't work out how to incorporate 2 design choices needed (mode agnostic vs not)
- System status vs Banner: creating a net new component to distinguish some component usage purpose feels against component creation, and could be hard to extend beyond current offering. I'd think System status (use case) is a pattern, together with "edge to edge" advice which could be optional for certain products. Similar to file upload pattern v.s. static list component.

**Comment by bhoppers2008** [comment]
@origami-z proposal looks good.

1. `filled` or `solid` - how did you determine which to use? (Assuming 'filling' inside the border?)
2. Assume full language for appearance is `solid, filled, bordered, transparent`?
3. Edge-edge - how to support this going forwards? @pseys `corner` won't be visible as option in new library. Is it prop driven, since JPM Brand is rounded corners?

**Comment by origami-z** [comment]
>1. `filled` or `solid` - how did you determine which to use? (Assuming 'filling' inside the border?)
> 2. Assume full language for appearance is `solid, filled, bordered, transparent`?

Yes, 4 total options. I'm open to other language as well. 
To explain, we could add something to the glossary around the term. 
1. there is a hierarchy of "importance" - solid > filled > bordered > transparent
2. define each, e.g. solid - bold background and border color, filled - subtle background + bold border, bordered - no background + border, transparent - no background  no border

> 3. Edge-edge - how to support this going forwards? @pseys `corner` won't be visible as option in new library. Is it prop driven, since JPM Brand is rounded corners?

Being a pattern, it would be a custom component from BL, which means a small override of corner could be applied there...?

**Comment by mark-tate** [comment]
Mocca Goal: If we have capacity, add button in code (medium size change) and initial docs
Publish to Labs

**Comment by mark-tate** [comment]
Red Eye: Tokens to be defined by EOS @pseys

**Comment by bhoppers2008** [comment]
Status update required @dplsek @pseys

**Comment by pseys** [comment]
Apologies @bhoppers2008 this took a backseat EO last year while I focused on the new libraries. I'll pick it up in this sprint.

**Comment by dplsek** [comment]
@pseys to create tokens by end of sprint

**Comment by bhoppers2008** [comment]
Goal: End of sprint to have the tokens in place and specs finalised.
Ben/ Josh/ Paul/ Darrin to align on call (@pseys to set up call).

**Comment by pseys** [comment]
Met with @dplsek , @bhoppers2008 and @joshwooding earlier today to discuss progress. Although token names have been proposed, Darrin has also proposed extending the scope of the Button. 

- Alpha/Ghost button should support the same appearances as `button` i.e. Transparent, Bordered, Solid
- It still needs to have two variants to allow for placement on both a light and mid/dark background in light mode.
- Josh confirmed that we could add conditions to the Button component to allow it to support `appearance` and/or `sentiment`/`alpha options` – the alternative is to follow on with @Fercas123 work and provide it as a stand-alone 'ghost button' component
- Ben shared that the approach may impact other components as we may need 'alpha card' etc. in the future
- Paul raised the need to clean up 'legacy' tokens as the way we handle opacity is flawed and inefficient - have begun mapping replacements/deprecations to allow legacy and brand to run off the same set of palette tokens. Doing this will minimise the number of tokens/css files we have to manage as a team and also simplify our Figma libraries considerably.

I've created [#4572](https://github.com/jpmorganchase/salt-ds/issues/4572) as the first part of cleaning up the tokens – this and any subsequent similar issues won't slow or prevent the button from being released

**Comment by mark-tate** [comment]
Tecno Goal: 
Standalone button available in Labs
- connect to the right token and show appearances (paul/fernanada)
- 2 variants it needs to allow it to sit in dark backgrounds have naming assigned (paul)
- call to discuss buttons and make decision on 
  - we will need to extend button to avoid creating alpha versions of all components that use button

**Comment by mark-tate** [comment]
AlphaButton versus extending Button semantics is still being discussed

**Comment by mark-tate** [comment]
Ontrack to have a decision by EOS

**Comment by mark-tate** [comment]
dependent on Pokemon, will stay in PR until we have interpreted the Pokemon matrix
Can have a snapshot release created if required

**Comment by bhoppers2008** [comment]
Update: Alpha button remains on hold until styling approach is confirmed.
